torch/package/importer.py

```diff
@@ -128,13 +128,30 @@ def get_obj_info(obj):
 
         obj_module_name, obj_location, obj_importer_name = get_obj_info(obj)
         obj2_module_name, obj2_location, obj2_importer_name = get_obj_info(obj2)
-        msg = (
-            f"\n\nThe object provided is from '{obj_module_name}', "
-            f"which is coming from {obj_location}."
-            f"\nHowever, when we import '{obj2_module_name}', it's coming from {obj2_location}."
-            "\nTo fix this, make sure this 'PackageExporter's importer lists "
-            f"{obj_importer_name} before {obj2_importer_name}."
-        )
+        msg = ""
+        if obj_module_name != obj2_module_name:
+            msg += (
+                f"\n\nCan't pickle {obj2}: it's not the same object as {obj_module_name}."
+                f"\nInstead, we are importing {obj2_module_name}. "
+                f"\nUsually these errors occur when trying to package unpicklable objects "
+                f"like decorated functions."
+            )
+        if (obj_location != obj2_location) or (obj_importer_name != obj2_importer_name):
```

Comment:
should this be an elif? are these mutually exclusive?

Reply:
They can both technically happen, I just didn't want anything to be silent